pkg/backends/locust/locust.go

```diff
@@ -0,0 +1,211 @@
+package locust
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	coreV1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/hellofresh/kangal/pkg/core/helper"
+	loadTestV1 "github.com/hellofresh/kangal/pkg/kubernetes/apis/loadtest/v1"
+	clientSetV "github.com/hellofresh/kangal/pkg/kubernetes/generated/clientset/versioned"
+	"go.uber.org/zap"
+)
+
+var (
+	defaultImage    = "locustio/locust"
+	defaultImageTag = "latest"
+)
+
+// Locust enables the controller to run a loadtest using locust.io
+type Locust struct {
+	kubeClientSet      kubernetes.Interface
+	kangalClientSet    clientSetV.Interface
+	loadTest           *loadTestV1.LoadTest
+	logger             *zap.Logger
+	reportPreSignedURL *url.URL
+	masterResources    helper.Resources
+	workerResources    helper.Resources
+	podAnnotations     map[string]string
+}
+
+// SetDefaults mutates the LoadTest object to add default values to empty fields
+func (c *Locust) SetDefaults() error {
```

Comment:
This is a duplicated code, would it be better to extract it to the helper?

Reply:
I had to change a little bit, so now it's different